### ISSUE TITLE
(game management) get single page up that can display the contents of a BGG API Call (Owen, Daniel, Will)

### ISSUE THREAD

**Issue by willcravitz** [uchicago-cs/chigame#46]
In issue #45 we are researching and documenting how to use the BoardGameGeeks API to get information on board games. We need to test that our documentation is correct by implementing a page that displays data from an API call.

**Comment by frowenz** [comment]
Update: I created a page with a search bar where you can enter text and search for games on Board Game Geek that match the text exactly. Any matching games will appear with a title, a description and an image. I am not sure if we should make a pull request given that this is just a demo of the functionality of the BGG API which we are going to want to integrate into the create game page.

Also, I tried to get the multiple closest results to display. but non-exact results are returned in alphabetical order and sorting them on popularity ended up taking a long time. I am not sure if there is a way to do this efficiently. 

<img width="1075" alt="Screenshot 2023-10-29 at 22 31 44" src="https://github.com/uchicago-cs/chigame/assets/69015620/5306eac4-fe36-40f5-a0bc-ceeedb9cd071">

A good next step would be writing the code to extract more of the information we need from a specific game. In other words,  the attributing mentioned [here](https://github.com/uchicago-cs/chigame/issues/39).

**Comment by danielngira** [comment]
Update: I have added more information to the page, specifically the year the game was published, its publishers, and the max and min play time. Can someone direct me where to get info on the games' rules? I haven't figured that out yet
<img width="1300" alt="Screenshot 2023-10-30 at 1 56 37 AM" src="https://github.com/uchicago-cs/chigame/assets/81624602/400af336-071f-400b-9e32-38afb9ff27b6">

**Comment by frowenz** [comment]
Nice work Daniel. Looking at the BGG API page and a sample request, I don't think it is possible to get the rules.

**Comment by frowenz** [comment]
I have moved the related code to [this wiki page ](https://github.com/uchicago-cs/chigame/wiki/Games-~-BoardGameGeek-(BGG)-API-Demo-Page) and deleted the related branch in order to avoid dangling branches.

**Comment by danielngira** [comment]
I have reopened this issue because of an update I made to the search page and because @fkbad instructed us to improve on the closing comment. As for my update: I have updated the game search view to include information on the board game designers and artists. I also added this data to the game search template. This is in response to the need that rose up when making models for the game in Django on issue #92.

![Image](https://github.com/uchicago-cs/chigame/assets/81624602/426a46ec-b253-4fc1-817b-70367299c93d)

**Comment by danielngira** [comment]
In issue #45 we researched how to access information from BoardGameGeek’s API.  The [wiki page](https://github.com/uchicago-cs/chigame/wiki/Games-~-BoardGameGeek-(BGG)-API) @willcravitz  made on making API calls was really helpful in this issue. We made a single page that includes a search bar that can search for games. These searches make calls to BoardGameGeek’s API and display the game's image, title, description, year published, publishers, minimum playtime, maximum playtime, designers & artists. @frowenz designed the template page and search bar. He also implemented a `GameListView` and a `bgg_search_view` which included data for the game’s image, title, and description. I updated the game search view to include the year published, publishers, minimum playtime, maximum playtime, artists, and designers. This template page will be helpful when implementing the page for creating games. I’m closing the issue now as we have completed our task to the best of my knowledge.
